client/cpp/encoder.cc

```diff
@@ -53,7 +54,7 @@ static void CheckValidProbability(float prob, const char* var_name) {
   if (prob < 0.0f || prob > 1.0f) {
     log("%s should be between 0.0 and 1.0 inclusive (got %.2f)", var_name,
         prob);
-    assert(false);
+    throw std::out_of_range("prob");
   }
```

Comment:
Although this is an OSS project, we generally follow Google style guide, which frowns upon use of C++ exceptions.